src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -49,6 +59,41 @@ public override void AttachToDependencyGraph(DependencyAnalyzerBase<NodeFactory>
 
             var compilerIdentifierNode = new CompilerIdentifierNode();
             CoreCLRReadyToRunHeader.Add(Internal.Runtime.ReadyToRunSectionType.CompilerIdentifier, compilerIdentifierNode, compilerIdentifierNode);
+
+            CoreCLRReadyToRunRuntimeFunctionsTable = new CoreCLRReadyToRunRuntimeFunctionsTableNode(Target);
+            graph.AddRoot(CoreCLRReadyToRunRuntimeFunctionsTable, "ReadyToRunRuntimeFunctionsTable is always generated");
```

Comment:
Thanks for pointing that out - at some point it seemed to not work but it was apparently due to a different bug, I have just deleted all the explicit AddRoot calls and R2RDump continues to work fine which is a clear indication that the header tables are still there as Amy uses the Directory [] operator so that the tool requires all these table to be available in the header.

Reply:
Hmm, I don't know why I keep writing "Directory" instead of "Dictionary", sorry about that :-).